js/dashboard-controls.js

```diff
@@ -0,0 +1,223 @@
+        var selector = "." + parent.attr("class").replace(/ /g, ".");
+        if (localStorage.defaultDataView == selector) {
+            // Double tap to turn off.
+            delete localStorage.defaultDataView;
+            setTimeout(function() {
+                parent.find("input").prop("checked", false);
+            }, 10);
+        } else {
+            // Set new default.
+            localStorage.defaultDataView = selector;
+        }
+    });
+
+    $('#leafwetness-button').click(function() {
+        $.ajax({
+            url: backendIpAddress + 'env_data',
+            data: {
+              vineyard_id: 1,
+              env_variable: 'leafwetness'
+            },
+            dataType: 'json',
+            type: "GET"
+        }).done(function(json) {
+            if (json.env_data) {
+                var envData = parseEnvData(json.env_data, 'lw');
+                map.removeLayer(idw);
+                idw = L.idwLayer(envData, idwOptions).addTo(map);
+            } else {
+                window.location.href = "dashboard.html";
+                /* TODO: add error handing message*/
+            }
+        }).fail(function() {
+            window.location.href = "dashboard.html";
+            /* TODO: add error handing message*/
+        });
+    });
+
+    $('#temperature-button').click(function() {
+        $.ajax({
+            url: backendIpAddress + 'env_data',
+            data: {
+              vineyard_id: 1,
+              env_variable: 'temperature'
+            },
+            dataType: 'json',
+            type: "GET"
+        }).done(function(json) {
+            if (json.env_data) {
+                var envData = parseEnvData(json.env_data, 'tmp');
+                map.removeLayer(idw);
+                idw = L.idwLayer(envData, idwOptions).addTo(map);
+            } else {
+                window.location.href = "dashboard.html";
+                /* TODO: add error handing message*/
+            }
+        }).fail(function() {
+            window.location.href = "dashboard.html";
+            /* TODO: add error handing message*/
+        });
+    });
+
+    $('#humidity-button').click(function() {
+        $.ajax({
+            url: backendIpAddress + 'env_data',
+            data: {
+              vineyard_id: 1,
+              env_variable: 'humidity'
+            },
+            dataType: 'json',
+            type: "GET"
+        }).done(function(json) {
+            if (json.env_data) {
+                var envData = parseEnvData(json.env_data, 'hum');
+                map.removeLayer(idw);
+                idw = L.idwLayer(envData, idwOptions).addTo(map);
+            } else {
+                window.location.href = "dashboard.html";
+                /* TODO: add error handing message*/
+            }
+        }).fail(function() {
+            window.location.href = "dashboard.html";
+            /* TODO: add error handing message*/
+        });
+    });
+
+    // $('#menu-leafwetness').click(function() {
```

Comment:
Commented code should be removed, at least for a release. It's in the commit history if we need it.